packages/models-library/src/models_library/services_resources.py

```diff
@@ -22,4 +40,82 @@ def ensure_limits_are_equal_or_above_reservations(cls, values):
         return values
 
 
-ServiceResources = DictModel[ResourceName, ResourceValue]
+Resources = DictModel[ResourceName, ResourceValue]
```

Comment:
1.  ``Resources`` is a very generic name (even banned in some standards as name https://google.aip.dev/122). In this case what you have is a map container of generic resources. I propose to rename this ``ResourcesDict``
2. Why did you choose ``DictModel[ResourceName, ResourceValue]`` vs ``dict[ResourceName, ResourceValue]``?? IMO the latter should be in general the preferred.

Reply:
1. 👍 
2. because `DictModel` is derived from a `Pydantic` model and has all the convenience of parsing and serialisation a such a model brings.

Reply:
More thinking regarding `2.`: will add more below.

Reply:
with 2. the advantages that you are listing you can still have them by using ``parse_obj_as(dict[ResrourceName, ResourceValue, obj) ``